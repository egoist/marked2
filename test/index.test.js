import marked from '../src'

test('headings', () => {
  const html = marked(`
# hello

# hello

## hi there
  `)

  expect(html).toMatchSnapshot()
})

test('table', () => {
  const html = marked(`
|foo|bar|
|---|---|
|foo|bar|
  `)

  expect(html).toMatchSnapshot()
})

test('links', () => {
  const html = marked(`
[a](b)
`)

  expect(html).toMatchSnapshot()

  const html2 = marked(
    `
[a](b)
  `,
    { linksInNewTab: true }
  )

  expect(html2).toMatchSnapshot()

  const html3 = marked(
    `
[a](b)
  `,
    { linksInNewTab: () => true }
  )

  expect(html3).toMatchSnapshot()

  const html4 = marked(
    `
[a](b)
  `,
    { linksInNewTab: () => false }
  )

  expect(html4).toMatchSnapshot()
})

test('data-line', () => {
  const html = marked(`
\`\`\`js{1,2,3,5-10}
hi
\`\`\`

\`\`\`css
hello
\`\`\`
`)

  expect(html).toMatchSnapshot()
})

test('inline-code', () => {
  const html = marked(`
Paragraph with \`some inline code\`.
  `,
    { codeSpanHighlighted: true }
  )

  expect(html).toMatchSnapshot()
})
