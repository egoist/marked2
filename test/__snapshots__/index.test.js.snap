// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data-line 1`] = `
"<pre data-line=\\"1,2,3,5-10\\"><code class=\\"lang-js\\">hi
</code></pre>
<pre><code class=\\"lang-css\\">hello
</code></pre>
"
`;

exports[`default inline code 1`] = `
"<p>Paragraph with <code>some inline code</code>.</p>
"
`;

exports[`headings 1`] = `
"<h1 id=\\"hello\\">hello</h1>
<h1 id=\\"hello-1\\">hello</h1>
<h2 id=\\"hi-there\\">hi there</h2>
"
`;

exports[`inline code with classes 1`] = `
"<p>Paragraph with <code class=\\"lang-markup\\">some inline code</code>.</p>
"
`;

exports[`links 1`] = `
"<p><a href=\\"b\\">a</a></p>
"
`;

exports[`links 2`] = `
"<p><a href=\\"b\\" target=\\"_blank\\">a</a></p>
"
`;

exports[`links 3`] = `
"<p><a href=\\"b\\" target=\\"_blank\\">a</a></p>
"
`;

exports[`links 4`] = `
"<p><a href=\\"b\\">a</a></p>
"
`;

exports[`table 1`] = `
"<table>
<thead>
<tr>
<th>foo</th>
<th>bar</th>
</tr>
</thead>
<tbody>
<tr>
<td>foo</td>
<td>bar</td>
</tr>
</tbody>
</table>
"
`;
